internal/pkg/argocd/argocd.go

```diff
@@ -170,26 +172,90 @@ func generateDiffOfAComponent(ctx context.Context, componentPath string, prBranc
 	}
 	foundApps, err := appIf.List(ctx, &appLabelQuery)
 	if err != nil {
-		log.Errorf("Error listing ArgoCD applications: %v", err)
-		componentDiffResult.DiffError = err
-		return componentDiffResult
+		return nil, fmt.Errorf("Error listing ArgoCD applications: %v", err)
+
 	}
 	if len(foundApps.Items) == 0 {
-		componentDiffResult.DiffError = fmt.Errorf("No ArgoCD application found for component path %s(repo %s), used this label selector: %s", componentPath, repo, labelSelector)
+		return nil, fmt.Errorf("No ArgoCD application found for component path sha1 %s(repo %s), used this label selector: %s", componentPathSha1, repo, labelSelector)
+	}
+
+	// we expect only one app with this label and repo selectors
+	return &foundApps.Items[0], nil
+}
+
+// This is the default method to find an ArgoCD application by the manifest-generate-paths annotation.
+// It assume the ArgoCD (optional) manifest-generate-paths annotation is set on all relevant apps.
+// Notice that this method include a full list all ArgoCD applications in the repo, this could be a performance issue if there are many apps in the repo.
+func findArgocdAppByManifestPathAnnotation(ctx context.Context, componentPath string, repo string, appIf application.ApplicationServiceClient) (app *argoappv1.Application, err error) {
+	// argocd.argoproj.io/manifest-generate-paths
+	appQuery := application.ApplicationQuery{
+		Repo: &repo,
+	}
+	// AFAIKT I can't use standard grpc instrumentation here, since the argocd client abstracts to much (including the choice between Grpc and Grpc-web)
```

Comment:
small typo:
```suggestion
	// AFAIKT I can't use standard grpc instrumentation here, since the argocd client abstracts too much (including the choice between Grpc and Grpc-web)
```